UI/resolution_picker.js

```diff
@@ -11,6 +12,11 @@ class ResolutionPicker {
      */
     constructor(resolution_select_id) {
         this._selector = document.getElementById(resolution_select_id);
+        // We capture when the user changes the resolution so we can update the scene
+        addEventListener("change", (e) => {
```

Comment:
When adding an event listener you have to add it to something specific, otherwise you're adding it to every single input element on the web page. With this code, if you change the dates, it refreshes the resolution, if you change opacity it refreshes the resolution etc.

```suggestion
        this._selector.addEventListener("change", (e) => {
```